ee/control/consumers/notificationconsumer/notify_consumer.go

```diff
@@ -28,14 +29,15 @@ type NotificationConsumer struct {
 
 // The desktop runner fullfils this interface -- it exists for testing purposes.
 type userProcessesRunner interface {
-	SendNotification(title, body string) error
+	SendNotification(title, body, actionUri string) error
```

Comment:
I think you're right that I should be passing around `notification` at this point, as we add more features to notifications. I hadn't before because it was a bit of a pain figuring out where to define it to avoid circular imports, but I may just need to do it once for the notification consumer and once for the notifier side.

Reply:
I don't think you can define it twice. But I think you could create a `types` somewhere if you needed to

Reply:
It's already defined twice -- once as `notificationconsumer.notification` (unexported, on the launcher root side) and once as `notify.Notification` (exported, on the desktop process side, now used in both `notify` and `server`). The desktop process side doesn't need the ID, ValidUntil, or SentAt fields, so I don't mind that it's split up -- but I think moving it to a `types` is the way to go, the dependency chain on the launcher root side is not working out well for doing it otherwise 🙂 .

Reply:
Never mind! I think the refactor I did in this PR and at the end of the last one made this much easier than I realized. There's now only 1 definition and we're passing `notification`s around everywhere.